src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java

```diff
@@ -288,12 +294,21 @@ private void checkLambda(DetailAST ast, DetailAST currentStatement) {
         }
     }
 
+    /**
+     * Checks that given node is a resource.
+     *
+     * @param ast semicolon to check
+     * @return true if node is a resource
+     */
     private static boolean isResource(DetailAST ast) {
-        return ast != null
-            && (ast.getType() == TokenTypes.RESOURCES
-                 || ast.getType() == TokenTypes.RESOURCE_SPECIFICATION);
+        return TokenUtil.isOfType(ast, TokenTypes.RESOURCES, TokenTypes.RESOURCE_SPECIFICATION);
```

Comment:
this one is not required, but just added it as a small enhancement

Reply:
@strkkk , please move this in separate commit, "minor: refactor ....." to keep it off javadoc beutification

Reply:
I removed this